src/application/shell/command/Ps.cpp

```diff
@@ -0,0 +1,38 @@
+#include "lib/libc/printf.h"
+#include "Ps.h"
+#include "kernel/thread/Scheduler.h"
+#include "kernel/thread/Thread.h"
+
+Ps::Ps(Shell &shell) : Command(shell) {
+
+};
+
+
+/*Driver code of ps command*/
+void Ps::execute(Util::Array<String> &args) {
+    /*Parses the help text for --help flag*/
+    Util::ArgumentParser parser(getHelpText(), 1);
+    if(!parser.parse(args)) {
+        stderr << args[0] << ": " << parser.getErrorString() << endl;
+        return;
+    }
+
+    /*Creates an instance of scheduler to access all the processes and counts the number of active processes*/
+    printf("Total Processes : %d\n",Kernel::Scheduler::getInstance().getThreadCount());
+    printf("-------------------------------\n");
+    printf("PID   Process Name\n");
+    printf("-------------------------------\n");
+    /*For each ready queue(based on priority order) we get the process info of all the processes in the queue*/
+    for (const auto &queue : Kernel::Scheduler::getInstance().readyQueues) {
```

Comment:
Please do not expose readyQueues as a public variable. If you need to access it, implement a getter-method.